### ISSUE TITLE
stdapi download language error?

### ISSUE THREAD

**Issue by kainpark7894** [rapid7/metasploit-framework#7082]
windows console metasploit

download c:/한글/한글.txt (korean language file download)

stdapi_fs_stat error

why?

![error](https://cloud.githubusercontent.com/assets/18137745/16677571/568a7dda-4512-11e6-80ea-c482a5b1af10.png)

english file no error

**Comment by kainpark7894** [comment]
![cmd](https://cloud.githubusercontent.com/assets/18137745/16677365/e89ae1f4-450f-11e6-8c84-d5ca77535ab2.png)

korean language path move,upload,download -> stdapi error

but no metasploit  windows cmd.exe korean path no error

chcp error?

unicode path error?

**Comment by egypt** [comment]
@bcook-r7 Didn't you do some work on unicode filesystem stuff? Is this related?

**Comment by bwatters-r7** [comment]
This is odd....  There could be more than one thing here.
I am sorry if the Korean below is incorrect; my Korean is worse than my Ruby.

## TL;DR
(1) I could not re-create exactly the above scenario
(2) I Could reliably get the above error with any Korean filename containing a space.
(3) Korean Notepad saves files as ANSI, so the 'cat' command looks broken at first.  When I specified saving the file in UTF-8, it displayed contents fine.
(4) Windows 7 cmd windows are horrible for unicode I/O.

## Full Report
I threw together a Korean language Windows 7x64 VM and connected with an x86 meterpreter exe through a reverse_tcp session.  I ran msfconsole on Ubuntu, so my results might be different.  Windows cmd and Unicode can be problematic.  When I started, I assumed that was the problem, and that it would work fine on Linux.
### msf vm:
```
Linux ubuntu 4.4.0-31-generic #50-Ubuntu SMP Wed Jul 13 00:07:12 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

### meterpreter VM:
```
meterpreter > sysinfo
Computer        : WIN-6PNJA50ADA7
OS              : Windows 7 (Build 7601, Service Pack 1).
Architecture    : x64 (Current Process is WOW64)
System Language : ko_KR
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x86/win32
```

I created three files:
```
meterpreter > ls
Listing: C:\Users\bwatters\Downloads
====================================

Mode              Size  Type  Last modified              Name
----              ----  ----  -------------              ----
100666/rw-rw-rw-  282   fil   2016-07-28 14:43:52 -0700  desktop.ini
100666/rw-rw-rw-  29    fil   2016-07-29 07:07:33 -0700  새 파일.txt
100666/rw-rw-rw-  29    fil   2016-07-29 07:09:51 -0700  새파일.txt
100666/rw-rw-rw-  29    fil   2016-07-29 07:08:35 -0700  파일.txt

```
I could not reproduce the error above with 파일.txt
```
meterpreter > download 파일.txt
[*] downloading: 파일.txt -> 파일.txt
[*] download   : 파일.txt -> 파일.txt
meterpreter > 
```

Verify:
```
meterpreter > cat 파일.txt

�ȳ��ϼ��� , ����meterpreter >
```
That is bad....
It was only because the encoding is ANSI.  
I made a new file encoded in UTF-8 to verify:

```
meterpreter > cat 새 파일.txt
[-] stdapi_fs_stat: Operation failed: The system cannot find the file specified.
meterpreter > download 새 파일.txt
[-] stdapi_fs_stat: Operation failed: The system cannot find the file specified.
meterpreter > 
```
That is worse.  After thinking about the differences, I created another file without a space in the name and tried again.
```
meterpreter > download 새파일.txt
[*] downloading: 새파일.txt -> 새파일.txt
[*] download   : 새파일.txt -> 새파일.txt
meterpreter > cat 새파일.txt

안녕하세요 , 세계

```
I created more files to test.....
```
meterpreter > ls
Listing: C:\Users\bwatters\Downloads
====================================

Mode              Size  Type  Last modified              Name
----              ----  ----  -------------              ----
100666/rw-rw-rw-  282   fil   2016-07-28 14:43:52 -0700  desktop.ini
100666/rw-rw-rw-  29    fil   2016-07-29 07:34:11 -0700  다른 파일.txt
100666/rw-rw-rw-  29    fil   2016-07-29 07:33:41 -0700  다른.txt
100666/rw-rw-rw-  29    fil   2016-07-29 07:34:19 -0700  다른파일.txt
100666/rw-rw-rw-  29    fil   2016-07-29 07:07:33 -0700  새 파일.txt
100666/rw-rw-rw-  29    fil   2016-07-29 07:09:51 -0700  새파일.txt
100666/rw-rw-rw-  19    fil   2016-07-29 07:15:30 -0700  파일.txt

meterpreter >
```

Results:
```
meterpreter > download 다른 파일.txt
[-] stdapi_fs_stat: Operation failed: The system cannot find the file specified.
meterpreter > download 다른.txt
[*] downloading: 다른.txt -> 다른.txt
[*] download   : 다른.txt -> 다른.txt
meterpreter > download 다른파일.txt
[*] downloading: 다른파일.txt -> 다른파일.txt
[*] download   : 다른파일.txt -> 다른파일.txt
meterpreter > download 새 파일.txt
[-] stdapi_fs_stat: Operation failed: The system cannot find the file specified.
meterpreter > download 새파일.txt
[*] downloading: 새파일.txt -> 새파일.txt
[*] skipped    : 새파일.txt -> 새파일.txt
meterpreter > download 파일.txt
[*] downloading: 파일.txt -> 파일.txt
[*] skipped    : 파일.txt -> 파일.txt
meterpreter > 
```

I am not sure if I was able to contribute anything to the original problem, but I think the next step in either case is to look at how msf passes filenames to meterpreter.  At minimum, it would be a good idea to print the filename it tried to find back out with the error message, though I am not sure how hard it will be.  I will take a look.

**Comment by egypt** [comment]
Files with spaces need to be quoted. Does your `download 다른 파일.txt` work if
you quote the filename?

**Comment by bwatters-r7** [comment]
Doh!
